examples/pump-app/cc13x2x7_26x2x7/main/AppTask.cpp

```diff
@@ -146,12 +166,12 @@ int AppTask::Init()
     }
 
 #if CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
-    chip::app::DnssdServer::Instance().SetExtendedDiscoveryTimeoutSecs(EXTENDED_DISCOVERY_TIMEOUT_SEC);
+    DnssdServer::Instance().SetExtendedDiscoveryTimeoutSecs(EXTENDED_DISCOVERY_TIMEOUT_SEC);
 #endif
 
     // Init ZCL Data Model and start server
     PLAT_LOG("Initialize Server");
-    chip::Server::GetInstance().Init();
+    Server::GetInstance().Init();
```

Comment:
Pre-existing, but file a followup please about fixing the fact that this is racing against the event loop that's already been started and can lead to data corruption?

Reply:
Thanks for the heads-up on these things. I have been looking into how the main.cpp and the AppTask.cpp handles all these initialization things. Comparing between projects, and I can't seem to find the know-good. Is there some guideline as to how this is to be done?

Reply:
@tlykkeberg-grundfos I just moved starting the CHIP thread to the end of the initialization code in nrfconnect examples, see https://github.com/project-chip/connectedhomeip/pull/15387.

Reply:
@Damian-Nordic thanks for the hint. I have also looked into your lock-app code ;-) I have an extra issue thou. The TI people have put a call to `ThreadStackMgrImpl().StartThreadTask()` into the Init sequence, which I can't seem to find in any other companies Init sequence. I have also tried to remove it, but this causes the Thread Stack to never run - it seems.

Reply:
When it comes to nRF Connect, OpenThread's thread is controlled by Zephyr, so we don't need to do it on the application side.

Reply:
I will change this in another PR later on, since we are having issues with rebooting a Matter device after it has been commissioned. This stalls sometimes, which we need to get fixed.

I will just resolve this conversation for now.